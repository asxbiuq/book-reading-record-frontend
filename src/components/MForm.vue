<template>
  <form>
    <slot name="above_label"></slot>
    <!-- 表单标题 -->
    <label>{{ formLabel }}</label>
    <slot name="under_label"></slot>
    <slot></slot>
    <slot name="above_button"></slot>
    <!-- 添加按钮 -->
    <div v-if="!isPending">
      <button>{{ btnName }}</button>
    </div>
    <div v-else>
      <button disabled>加载中...</button>
    </div>
    <slot name="under_button"></slot>
  </form>
</template>

<script setup>
const props = defineProps(['formLabel', 'btnName','isPending'])
</script>

<style scoped>
@import 'styles/utilities.css';
@import 'styles/components/btn.css';
@tailwind utilities;
@tailwind components;
</style>
