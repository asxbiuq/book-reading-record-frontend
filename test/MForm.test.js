import { flushPromises, mount } from '@vue/test-utils'
import MForm from 'components/MForm.vue'

test('mount component', async () => {
  expect(MForm).toBeTruthy()

  const wrapper = mount(MForm, {
    props: {
      formLabel: '表单',
      btnName: '按钮',
    },
  })

  expect(wrapper.text()).toContain('表')
  expect(wrapper.html()).toMatchSnapshot()

  // await wrapper.get('button').trigger('click')

  // expect(wrapper.text()).toContain('4 x 3 = 12')

  // await wrapper.get('button').trigger('click')

  // expect(wrapper.text()).toContain('4 x 4 = 16')
})
