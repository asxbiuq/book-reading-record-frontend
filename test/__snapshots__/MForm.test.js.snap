// Vitest Snapshot v1

exports[`mount component 1`] = `
"<form>
  <!-- 表单标题 --><label>表单</label><!-- 添加按钮 --><button>按钮</button>
</form>"
`;
